### DISCUSSION TITLE
Mouse lock across experiment?

### DISCUSSION THREAD

**Discussion by paulzerr** [jspsych/jsPsych#1094] [Q&A]
Hi all,

I'm using a continuous report paradigm where the subject moves the mouse to rotate a probe bar. However, when the mouse pointer reaches the edge of the screen the rotation is stuck and participants find that frustrating.

To solve this I would like to implement the fantastic [Mouse Lock](https://developer.mozilla.org/en-US/docs/Web/API/Pointer_Lock_API) functionality.  [See a demo here.](https://mdn.github.io/dom-examples/pointer-lock/). 

I'm unsure how that can work within jspsych, because I want to avoid that participants first have to click on a canvas every time and see that message at the top of the screen "your mouse is now locked, press escape to...", but instead have them do that only once at the start of the experiment. 

If I understand correctly, a canvas, or anything on the screen will be wiped on every 'trial'. Is that correct? Do you have any ideas what I could do here?

Thanks much!

**Comment by becky-gilbert** [comment]
Hi @rainfern! I didn't know about mouse lock - looks very cool! Have you tried creating multiple "mouse lock" trials and running them on a server? I'm wondering whether or not the participant will be asked permission every time. With the audio recording mic permissions, I know that the browser will ask on every trial when the HTML file is opened directly in the browser, but once the file is hosted by a server, the browser usually just asks the first time (this depends on the browser and settings). So I'm wondering if it might be similar for mouse lock. I noticed in docs that it says:

> If a user has exited pointer lock via the default unlock gesture, or pointer lock has not previously been entered for this document, an event generated as a result of an engagement gesture must be received by the document before requestPointerLock will succeed.

So I _think_ this means that once the user has previously entered mouse lock on the same webpage, they won't be asked again the next time (unless they also exited pointer lock via the escape key). Have you checked this already?
